src/lib/Simplify.hs

```diff
@@ -698,38 +695,49 @@ simplifyOp op = case op of
         return $ PairVal (head $ toList val) $
           Record otherItemTys $ toList rest
       _ -> error "not a record"
-  VariantLift leftTys@(LabeledItems litems) right -> getType right >>= \case
+  VariantLift leftTys right -> getType right >>= \case
     VariantTy (NoExt rightTys) -> do
-      let fullRow = NoExt $ leftTys <> rightTys
+      let fullTys = leftTys <> rightTys
+      let fullRow = NoExt fullTys
+      let fullLabels = toList $ reflectLabels fullTys
       let labels = toList $ reflectLabels rightTys
       -- Emit a case statement (ordered by the arg type) that lifts the type.
       buildCase right (VariantTy fullRow) \caseIdx [v] -> do
-          let (label, i) = labels !! caseIdx
-          let idx = case M.lookup label litems of Nothing  -> i
-                                                  Just tys -> i + length tys
-          let fullRow' = fromExtLabeledItemsE $ sink $ ExtLabeledItemsE fullRow
-          return $ Variant fullRow' label idx v
+        -- TODO: This is extremely slow! Optimize this! Use a prefix sum?
```

Comment:
Could you spell out what's slow and how to make it better at one more level of detail?

Reply:
Done.